File: Search Algorithms/Linear Search/LinearSearch.cs
//Linear Search Algorithm 
public static int LinearSearch(int[] data, int item)
{
 int N=data.Length;
 for (int i=0; i < N; i++)
	if (data[i] == item)
	   return i;
 return -1;
}

//Example 
static void Main(string[] args)
{
Console.WriteLine("Please Enter Some Integer values separate them by using ,");
string EnteredIntegers = Console.ReadLine();
int position = 0;
bool isFound = false;
string[] strIntegers = EnteredIntegers.Split(',');

if (strIntegers.Length <= 0)
{
	Console.WriteLine("Invalid InPut");
	return;
}

Console.WriteLine("Please Enter the Number You Want To Search");

int searchNum = Convert.ToInt32(Console.ReadLine());


	for (int i = 0; i < strIntegers.Length; i++)
	{
		if (Convert.ToInt32(strIntegers[i]).Equals(searchNum))
		{
			position = i;
			isFound = true;
		}
	}

if (isFound)
{
	Console.WriteLine("Your Search Value :" + searchNum.ToString() + " Is at position :" + position.ToString());
}
else
{
	Console.WriteLine("No Items Match Your Selection");
}

Console.ReadLine();
}

File: Search Algorithms/Linear Search/LinearSearch.c
#include <stdio.h>
#include <stdlib.h>

int LinearSearch(int array[],int sk,int size)
{
 int x;

 for(x=0;x<size;x++)
 {
     if(array[x]==sk)
     {
         break;
     }
 }
 return x;

}

int main()
{
 int sk,size,x,result;

 printf("Enter the Size of the Array ");
 scanf("%d",&size);
 int array[size];

 for(x=0;x<size;x++)
 {
     printf("Enter Value %d to Array ",x+1);
     scanf("%d",&array[x]);
 }

 printf("Enter the Search Key ");
 scanf("%d",&sk);

 result=LinearSearch(array,sk,size);

 if(result==size)
    printf("The Value is Not Found");
 else
 printf("The Index of the Value is %d ",result);

    return 0;
}
